firebase-firestore/ktx/src/test/kotlin/com/google/firebase/firestore/ktx/FirestoreMapDecoderTests.kt

```diff
@@ -0,0 +1,226 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore.ktx
+
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.firestore.DocumentReference
+import com.google.firebase.firestore.assertThrows
+import com.google.firebase.firestore.documentReference
+import com.google.firebase.firestore.ktx.serialization.decodeFromMap
+import kotlinx.serialization.Serializable
+import org.junit.BeforeClass
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
+class FirestoreMapDecoderTests {
+
+    companion object {
+        lateinit var firestoreDocument: DocumentReference
+
+        @BeforeClass
+        @JvmStatic
+        fun generateDocRef() {
+            firestoreDocument = documentReference("abc/1234")
+        }
+    }
```

Comment:
What's the reason this is created here instead of in a regular `@Before` method?